src/Php84/bootstrap.php

```diff
@@ -15,6 +15,14 @@
     return;
 }
 
+if (function_exists('curl_version') && curl_version()['version'] >= 0x074200) { // libcurl >= 7.66.0
+    define('CURL_HTTP_VERSION_3', 30);
+
+    if (curl_version()['version'] >= 0x075800) { // libcurl >= 7.88.0
```

Comment:
I'm a bit annoyed by this slow path, how about not doing the check and always declare the const next to the previous one?
Accuracy of the polyfilling on this aspect doesn't look more important than raw perf here to me.

Reply:
Things add up: there are many polyfills to load: if we're not careful with each of them, this might make things slow without us realizing. For now, most (all?) checks are resolved at compile-time so they're free.

Reply:
It's not the slowest call, but `curl_version()` has a lot of information mostly by checking bitmasks for protocol and feature support. We only need the version here, so yeah I agree it makes sense to use something like `defined('CURLOPT_UPLOAD_BUFFERSIZE')` that also implicitly ensures libcurl >= 7.62

Reply:
```suggestion
    if (defined('CURLOPT_SSH_HOST_PUBLIC_KEY_SHA256')) { // libcurl >= 7.80.0 (7.88 would be better but is slow to check)
```

Reply:
LGTM